passkey-types/src/webauthn/attestation.rs

```diff
@@ -569,13 +791,78 @@ pub enum ClientDataType {
     /// Serializes to the string `"webauthn.get"`
     #[serde(rename = "webauthn.get")]
     Get,
+
+    /// Serializes to the string `"payment.get"`
+    #[serde(rename = "payment.get")]
+    PaymentGet,
+
+    /// Serialize to an empty string `""`
+    #[serde(rename = "")]
+    Empty,
+}
+
+impl fmt::Display for ClientDataType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let renamed = match self {
+            ClientDataType::Create => "webauthn.create",
+            ClientDataType::Get => "webauthn.get",
+            ClientDataType::PaymentGet => "payment.get",
+            ClientDataType::Empty => "",
+        };
+        write!(f, "{}", renamed)
+    }
+}
+
+impl ClientDataType {
+    /// Helper function to determine if `ClientDataType` is empty
+    pub fn is_empty(&self) -> bool {
+        matches!(self, ClientDataType::Empty)
+    }
 }
 
 #[cfg(test)]
 mod tests {
-
+    use crate::webauthn::{ClientDataType, CollectedClientData};
     use super::CredentialCreationOptions;
 
+    // Normal client data from Chrome assertion
+    static CLIENT_DATA_BYTES: [u8; 134] = [
+        123, 34, 116, 121, 112, 101, 34, 58, 34, 119, 101, 98, 97, 117, 116, 104, 110, 46, 103,
+        101, 116, 34, 44, 34, 99, 104, 97, 108, 108, 101, 110, 103, 101, 34, 58, 34, 90, 69, 118,
+        77, 102, 108, 90, 68, 99, 119, 81, 74, 109, 97, 114, 73, 110, 110, 89, 105, 56, 56, 112,
+        120, 45, 54, 72, 90, 99, 118, 50, 85, 111, 120, 119, 55, 45, 95, 74, 79, 79, 84, 103, 34,
+        44, 34, 111, 114, 105, 103, 105, 110, 34, 58, 34, 104, 116, 116, 112, 58, 47, 47, 108, 111,
+        99, 97, 108, 104, 111, 115, 116, 58, 52, 48, 48, 48, 34, 44, 34, 99, 114, 111, 115, 115,
+        79, 114, 105, 103, 105, 110, 34, 58, 102, 97, 108, 115, 101, 125,
+    ];
```

Comment:
**Suggestion:** Instead of an opaque blob of bytes, it would be easier to maintain the string representation.

Reply:
I wanted to make sure it was the true byte representation of the clientDataJSON straight from the assertion response so that I knew the serialization for CollectedClientData was working properly, but happy to convert it to a string. Would you want `base64Url`?

Reply:
I was thinking more along the lines of

```rust
const CLIENT_DATA_BYTES: &[u8] =  r#"{"type":"webauthn.get","challenge":"ZEvMflZDcwQJmarInnYi88px-6HZcv2Uoxw7-_JOOTg","origin":"http://localhost:4000","crossOrigin":false}"#.as_bytes();
```

Like that we can edit it in place if needed.

Reply:
If I do a direct `as_bytes()` on this string, it will include any accidental utf-8 characters in the byte serialization so I'll make this string a constant and have the test functions convert it into the json byte representation